src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionAzureKeyVaultProvider.java

```diff
@@ -84,96 +95,126 @@ public String getName() {
+    SQLServerColumnEncryptionAzureKeyVaultProvider() throws SQLServerException {
+        setCredential(new ManagedIdentityCredentialBuilder().build());
     }
 
     /**
-     * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider with a callback function to authenticate to AAD. This
-     * is used by KeyVaultClient at runtime to authenticate to Azure Key Vault.
+     * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider to authenticate to AAD. This is used by KeyVault
+     * client at runtime to authenticate to Azure Key Vault.
+     *
+     * @param clientId
+     *        Identifier of the client requesting the token.
      * 
-     * @param authenticationCallback
-     *        - Callback function used for authenticating to AAD.
      * @throws SQLServerException
      *         when an error occurs
      */
-    public SQLServerColumnEncryptionAzureKeyVaultProvider(
-            SQLServerKeyVaultAuthenticationCallback authenticationCallback) throws SQLServerException {
-        if (null == authenticationCallback) {
+    SQLServerColumnEncryptionAzureKeyVaultProvider(String clientId) throws SQLServerException {
+        if (null == clientId || clientId.isEmpty()) {
             MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_NullValue"));
-            Object[] msgArgs1 = {"SQLServerKeyVaultAuthenticationCallback"};
+            Object[] msgArgs1 = {"Client ID"};
             throw new SQLServerException(form.format(msgArgs1), null);
         }
-        credentials = new KeyVaultCredential(authenticationCallback);
-        RestClient restClient = new RestClient.Builder(new OkHttpClient.Builder(), new Retrofit.Builder())
-                .withBaseUrl(baseUrl).withCredentials(credentials).withSerializerAdapter(new AzureJacksonAdapter())
-                .withResponseBuilderFactory(new AzureResponseBuilder.Factory()).build();
-        keyVaultClient = new KeyVaultClient(restClient);
+        setCredential(new ManagedIdentityCredentialBuilder().clientId(clientId).build());
     }
 
     /**
-     * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider to authenticate to AAD. This is used by
-     * KeyVaultClient at runtime to authenticate to Azure Key Vault.
+     * Constructs a SQLServerColumnEncryptionAzureKeyVaultProvider using the provided TokenCredential to authenticate to
+     * AAD. This is used by KeyVault client at runtime to authenticate to Azure Key Vault.
+     *
+     * @param tokenCredential
+     *        The TokenCredential to use to authenticate to Azure Key Vault.
      * 
      * @throws SQLServerException
      *         when an error occurs
      */
-    SQLServerColumnEncryptionAzureKeyVaultProvider() throws SQLServerException {
-        credentials = new KeyVaultCredential();
-        keyVaultClient = new KeyVaultClient(credentials);
+    public SQLServerColumnEncryptionAzureKeyVaultProvider(TokenCredential tokenCredential) throws SQLServerException {
```

Comment:
There is no tests for this constructor, please add a few tests. I could only find one test, which only tests bad AKV name.